### PR TITLE
Copy old mappings to _meta section

### PR DESCRIPTION
**Pull request by ywelsch**
For archival indices, where mappings might not be parseable by new ES versions anymore, copy the mapping to the `_meta/legacy_mappings` section.

Relates #81210

### CODE REVIEW

**Comment by ywelsch** [comment]
@elasticmachine run elasticsearch-ci/part-2 (unrelated test failure)

**Comment by ywelsch** [comment]
@elasticmachine update branch

**Comment by ywelsch** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-unix-sample

File: server/src/main/java/org/elasticsearch/snapshots/RestoreService.java [hunk 1]
```diff
@@ -1582,6 +1589,32 @@ public void clusterStateProcessed(ClusterState oldState, ClusterState newState)
...
+        Map<String, Object> loadedMappingSource = mappingMetadata.rawSourceAsMap();
+
+        // store old mapping under _meta/legacy-mappings
+        Map<String, Object> legacyMapping = new LinkedHashMap<>();
+        boolean sourceOnlySnapshot = snapshotIndexMetadata.getSettings().getAsBoolean("index.source_only", false);
+        if (sourceOnlySnapshot) {
+            // actual mapping is under "_meta" (but strip type first)
+            Object sourceOnlyMeta = mappingMetadata.sourceAsMap().get("_meta");
+            if (sourceOnlyMeta instanceof Map<?, ?> sourceOnlyMetaMap) {
+                legacyMapping.put("legacy-mappings", sourceOnlyMetaMap);
```
Comment (dnhatn) [inline server/src/main/java/org/elasticsearch/snapshots/RestoreService.java:1603]: Should we use `legacy_mappings` instead of `legacy-mappings`?

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks Yannick.

File: server/src/main/java/org/elasticsearch/snapshots/RestoreService.java (hunk 1 above)
Comment (ywelsch) [reply server/src/main/java/org/elasticsearch/snapshots/RestoreService.java]: fixed in 939bc51

**Comment by ywelsch** [comment]
Thanks @dnhatn!

**Pull request merged**